pandas/io/parquet.py

```diff
@@ -90,10 +90,20 @@ def _get_path_or_handle(
     path_or_handle = stringify_path(path)
     if is_fsspec_url(path_or_handle) and fs is None:
         fsspec = import_optional_dependency("fsspec")
+        if storage_options is None:
+            pa = import_optional_dependency("pyarrow")
+            pa_fs = import_optional_dependency("pyarrow.fs")
 
-        fs, path_or_handle = fsspec.core.url_to_fs(
-            path_or_handle, **(storage_options or {})
-        )
+            try:
+                fs_arrow = import_optional_dependency("fsspec.implementations.arrow")
+                fs, path_or_handle = pa_fs.FileSystem.from_uri(path)
+                fs = fs_arrow.ArrowFSWrapper(fs)
```

Comment:
Is there a reason we need to wrap it here? Are we using some fsspec APIs? From looking at the code, it seems we just pass this object through to pyarrow as `filesystem` keyword? In that case, we can (and should) pass the actual pyarrow filesystem .

(in pyarrow, we should probably unwrap such an fsspec filesystem that wraps a pyarrow filesystem, but at the moment we don't do that, we just see it as a generic fsspec filesystem)

Reply:
I was thinking if there is a future fix/enhancement that expects the filesystem to be fsspec-like that this would be convenient, but it isn't needed particularly now. We can add that if needed in the future

(I was under the impression that `ArrowFSWrapper` would be interpreted as an pyarrow filesystem since that's the backend from https://arrow.apache.org/docs/python/filesystems.html#using-arrow-filesystems-with-fsspec)

Reply:
> (I was under the impression that `ArrowFSWrapper` would be interpreted as an pyarrow filesystem since that's the backend from https://arrow.apache.org/docs/python/filesystems.html#using-arrow-filesystems-with-fsspec)

That section is for if you want to use a pyarrow filesystem in an fsspec context (for example for passing it to a function that expects an fsspec filesystem, or to use the richer API it provives). But in this case we are not in such a context, since we are passing the filesystem to a pyarrow function. 

It doesn't really say anything about then using that wrapper in a pyarrow context, but as mentioned above, we can certainly unwrap this inside pyarrow, avoiding the confusion / wrong expectation you just had (but so currently we don't do that yet).

Reply:
Opened https://github.com/apache/arrow/issues/34542